src/test/regress/sql/local_table_join.sql

```diff
@@ -335,6 +335,39 @@ ALTER TABLE local DROP column key3;
 ALTER TABLE local DROP column key1;
 SELECT COUNT(*) FROM distributed_table JOIN local ON distributed_table.value = 'text';
 
+--Issue 4678
+
+create table tbl (a int);
+select create_distributed_table('tbl', 'a');
+
+-- subplans work but we might skip the restrictions in them
+select 1 from (
+  select 1 from (
+    select 1 from
+      pg_catalog.pg_type,
+      lateral (
+        select 1 from tbl
+        where typdefault ~ null
+        limit 1) as subq_0
+    where (
+      select true from pg_catalog.pg_am
+      where typdefault > 'a'
+    )
+  ) as subq_1
+) as subq_2;
+
+-- sublink
+SELECT COUNT(*) FROM distributed_table JOIN postgres_table using(key)
```

Comment:
I thought this would be a sublink but debugging it doesn't show it as a sublink

Reply:
Maybe use subqueries in the target list to generate sublinks?

Reply:
what difference does it have compared to a subquery in WHERE clause?